### PR TITLE
Add new mix db vendors template

### PR DESCRIPTION
**Pull request by fuadagazade**

### CODE REVIEW

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/DesignTimeDbContextFactory.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                .Add(new CustomEnvironmentVariableConfigurationSource())
+                .Build();
+
+            if (IsCheckConnectionStringExists())
+            {
+                string connectionString = Environment.GetEnvironmentVariable("ConnectionStrings");
+
+                if (IsInjectDbCredentialsToConnectionString())
+                {
+                    #if SQLSERVER
```
Comment (Joseph1977) [inline src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/DesignTimeDbContextFactory.cs:61]: @fuadagazade why need this here? if it is the exact same and load from environment variable?

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+            app.UseSwagger(apiVersionDescriptionProvider, Configuration);
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+            });
+        }
+
+        private void ConfigureSqlServerContext(IServiceCollection services)
```
Comment (Joseph1977) [inline src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs]: @fuadagazade i do not like this duplication , you can use same funtion and the do the ifdef only for the line with the 
  services.AddDbContext

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi.IntegrationTests/Config.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            {
+                string connectionString = _configuration.GetValue<string>("ConnectionStrings");
+                #if SQLSERVER
+                if (IsInjectDbCredentialsToConnectionString())
+                {
+                    connectionString +=
+                        $";User Id={_configuration.GetValue<string>("AspNetCoreDbUserName")};Password='{_configuration.GetValue<string>("AspNetCoreDbPassword")}'";
+                }
+
+                builder.UseSqlServer(connectionString);
```
Comment (Joseph1977) [inline src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi.IntegrationTests/Config.cs:84]: @fuadagazade this is the only different line, so the #if def should be only here

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+        /// <summary>
+        /// Configures application services.
+        /// </summary>
+        /// <param name="services">Services.</param>
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+            services.AddCors();
+
+            #if SQLSERVER
```
Comment (Joseph1977) [inline src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs]: @fuadagazade no need 2 function one function and inside only the different lines

**Review by Joseph1977 — APPROVED** [review]
@fuadagazade please remove duplication in code where i added comments. 
if you have a different only in one line, the #ifdef should be only there and no need to duplicate all the section

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/DesignTimeDbContextFactory.cs (hunk 1 above)
Comment (fuadagazade) [reply src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/DesignTimeDbContextFactory.cs:61]: In the SQL Server ConnectionString contains User Id, in PostgreSQL this is Username

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi.IntegrationTests/Config.cs (hunk 3 above)
Comment (fuadagazade) [reply src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi.IntegrationTests/Config.cs:84]: In the SQL Server ConnectionString contains User Id, in PostgreSQL this is Username

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs (hunk 2 above)
Comment (fuadagazade) [reply src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs]: Changed

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs (hunk 4 above)
Comment (fuadagazade) [reply src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs]: Changed

**Comment by Joseph1977** [comment]
@parimal1987 pls review

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs [hunk 5]
```diff
@@ -61,11 +63,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddMvc();
             services.AddCors();
 
-            #if SQLSERVER
             ConfigureSqlServerContext(services);
```
Comment (devparimal1987) [inline src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs]: Change the method name to ConfigureDatabaseProviderContext

**Review by devparimal1987 — APPROVED** [review]
Approved with small comment

File: src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs (hunk 5 above)
Comment (fuadagazade) [reply src-env (mixdbvendors)/_MicroserviceTemplate_.WebApi/Startup.cs]: Done

**Pull request merged**